server/types/catalog.go

```diff
@@ -0,0 +1,190 @@
+// A service contains many versions. Each version contains 1 or more containers to deploy
+// Containers of a service are bound together, meaning if the service is stopped, all containers are stopped
+// Jobs (checkhealth) or commands can be executed on containers
+type CatalogService struct {
+	ID       bson.ObjectId                    `bson:"_id,omitempty" json:"id,omitempty"`
+	Name     string                           `bson:"name" json:"name"`
+	LogoPath string                           `bson:"logoPath" json:"logoPath"`
+	Versions map[string]CatalogServiceVersion `bson:"versions" json:"versions"`
+	Tags     []bson.ObjectId                  `bson:"tags" json:"tags"`
+	Created  time.Time                        `bson:"created" json:"created"`
+	Updated  time.Time                        `bson:"updated" json:"updated"`
+}
+
+// CatalogServiceVersion defines a version for a service from catalog. It contains the version number and all its metadata
+// A service version is composed of many containers
+type CatalogServiceVersion struct {
+	// False means this version should not be used anymore, then should not be displayed as new version to deploy
+	Active bool `bson:"active" json:"active"`
+	// The version number of the whole service. Is usually the version of the main container
+	Name string `bson:"name" json:"name"`
+	// Markdown description of what has changed in this version
+	Changelog string `bson:"changelog" json:"changelog"`
+	// Last compatible version number for this current version
+	// If empty, Docktor will not be able to upgrade automatically a previous service to this current service version
+	LastCompatibleVersion string `bson:"lastCompatVersion" json:"lastCompatVersion"`
+	// 1 or more container defining the service.
+	// e.g. a database and a web application are two containers for a single service
+	Containers []CatalogContainer `bson:"containers" json:"containers"`
+	Created    time.Time          `bson:"created" json:"created"`
+	Updated    time.Time          `bson:"updated" json:"updated"`
+}
+
+// CatalogContainer defines a container configuration inside a given catalog service
+// It contains
+type CatalogContainer struct {
+	ID bson.ObjectId `bson:"_id,omitempty" json:"id,omitempty"`
+	// A user-friendly name to distingish deployed containers inside a deployed service
+	Name string `bson:"name" json:"name"`
+	// Image version like 'registryname/imagename:tag'
+	Image string `bson:"image" json:"image"`
+	// Type of container (primary or sidekick), default to primary
+	Type ContainerType `bson:"type" json:"type"`
+	// Available commands to execute inside container
+	Commands Commands `bson:"commands" json:"commands"`
+	// URls to reach the container
+	URLs URLs `bson:"urls" json:"urls"`
+	// Scheduled job for healthcheck
+	HealthCheck HealthCheck `bson:"healthCheck" json:"healthCheck"`
+	// Default variables for given container
+	Variables Variables `bson:"variables" json:"variables"`
```

Comment:
En voyant ça ça me fait penser qu'on a pas pensé au support des [args](https://docs.docker.com/engine/reference/run/#cmd-default-command-or-options) au lancement de conteneur. Je pense que c'est important de les supporter car il y a pas mal de conteneurs dont la config passe par les args et pas par les variables d'environnement (exemple : [lancer un mongo officiel en mode sécurisé](https://hub.docker.com/_/mongo/), voir la section "Authentication and Authorization").

Reply:
Done